### ISSUE TITLE
Cursor snap points are weird and random, can't precisely select most regions

### ISSUE THREAD

**Issue by JC3** [audacity/audacity#3680]
### Bug description

The snap points for the cursor don't seem to follow any pattern, and are often unavailable at key locations.

As a consequence, it isn't reliably possible to select specific areas with any type of precision without mucking around without stored cursors and the transport menu.

### Steps to reproduce

1. Move the mouse around over the timeline in a project with multiple tracks / clips and observe the yellow snap lines.

### Expected behavior

Snap points at 0, the start and end of clips, and the start and end of labels, such as:

![image](https://user-images.githubusercontent.com/5651184/192030154-5fe8fbeb-7144-4b96-ac8e-9093569c7b70.png)

### Actual behavior

Snap points randomly chosen from start and end points of arbitrary items:

![image](https://user-images.githubusercontent.com/5651184/192029054-711e8625-638e-443b-a13e-d6835e09ea3e.png)

![snapsopt](https://user-images.githubusercontent.com/5651184/192029587-0d36b032-d200-4dae-a6b6-946c52a70f7e.gif)

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows

**Comment by petersampsonaudacity** [comment]
Testing on W10 with 3.2.0

I get snap lines at 

1) clip start or end
![image](https://user-images.githubusercontent.com/50205984/192136450-344c46ad-785e-4316-9631-ed73da1f9345.png)

2) Label start and end
![image](https://user-images.githubusercontent.com/50205984/192136485-726be54b-0afe-4eee-b2e7-77d314714bd3.png)

And I use these a LOT in my editing with Audacity

The one place you don't get a clip line with clips is when you have two adjoining clips and you hover over the clip join - and I suspect that is because of the "feature" that allows you to make a left click there to join the two clicks.  However if you hover the cursor in that time position in any other track then you will see a snap line at the junction of the adjacent clips.

**Comment by JC3** [comment]
> Testing on W10 with 3.2.0
> 
> I get snap lines at
> 
> 1. clip start or end
> 2. Label start and end

Here's the list of snap lines I did and didn't get in the OP (writing these in text is probably clearer then the image); numbering tracks top to bottom and clips left to right; expectation is "yes" for all:

| Object | Snap line at start? | Snap line at end? |
|-|-|-|
| Project | Yes (time=0) | **No** |
| Track 1, clip 1 | Yes | Inconsistent* |
| Track 1, clip 2 | **No** | **No** |
| Track 1, clip 3 | **No** | **No** |
| Track 2, clip 1 | Yes | **No** |
| Track 2, clip 2 | **No** | **No** |
| Track 2, clip 3 | **No** | **No** |
| Track 3, label "a" | **No** | Yes |
| Track 3, label "b" | **No** | **No** |
| Track 3, label "c" | Yes | **No** |
| Track 3, label "d" | **No** | Yes |

Note: The snap line at the end of track 1 clip 1 did not appear on initial mouse over; instead the horizontal resize (↔) mouse cursor appeared. After moving the mouse away and back, the snap line appeared, and continued to appear on all subsequent mouseovers. This occurs around timestamp 0:03 in the video.


> The one place you don't get a clip line with clips is when you have two adjoining clips and you hover over the clip join - and I suspect that is because of the "feature" that allows you to make a left click there to join the two clicks. However if you hover the cursor in that time position in any other track then you will see a snap line at the junction of the adjacent clips.

I don't have Audacity open right now to check but I do remember in 2.x, if you had the mouse over a clip boundary and the "join" cursor appeared, you could press Esc and the join cursor would go away and a snap line would appear (I think the status bar text said "press esc to cancel" when you got the join cursor). It was kind of weird. Not sure if 3.x still behaves that way.

**Comment by dozzzzer** [comment]
I'm observing the exact same behavior on both macOS and Windows builds: some of the clip starts/ends don't have snap lines, while I see them at the label starts/ends.
Here's an example:

https://user-images.githubusercontent.com/70063970/193007440-58c0154d-6e22-4bf2-815a-fb10e0c0f412.mp4

**Comment by petersampsonaudacity** [comment]
@dozzzzer  -  well that's most odd  -  as testing on 3.2.0 on W10 with a replica of your project I get snap lines at all the places I would expect to see them, including the two places where you don't get them in your video ...

![image](https://user-images.githubusercontent.com/50205984/193016810-16599c54-5ed4-4ddf-ac01-60cc8c347969.png)

**Comment by dozzzzer** [comment]
@petersampsonaudacity would you please try my project from the video 
[missing snap lines.aup3.zip](https://github.com/audacity/audacity/files/9674369/missing.snap.lines.aup3.zip)

**Comment by petersampsonaudacity** [comment]
@dozzzzer 

Yepper I get several missing snap lines with your project - hmmmmm

**Comment by dozzzzer** [comment]
@petersampsonaudacity yes, as soon as I start dragging the right edge of the clip, the snap line appears. As soon as I release it, it's gone again.

**Comment by JC3** [comment]
I was just going to post again here, too, to clarify that I still consistently see the issue even with non-contrived projects and add some maybe-helpful data points. 

So here, for example:

![ezgif com-gif-maker (6)](https://user-images.githubusercontent.com/5651184/193052344-73e99fde-9440-4f22-9aee-4a50358f0351.gif)

There, only the start + end of project snap lines show, as well as the snap line on the 0-length label. The clip and region label boundary lines do not appear.

Additionally, moving the start position of the label with the working snap line permanently killed the snap line on that label.

From this, and other things in this video and project, it *seems* like (these may not be true):

- Working vs broken isn't associated with the item itself (e.g. moving a working label made it broken)
- Zoom level does not have an effect
- Scroll position does not have an effect
- Restarting Audacity does not have an effect
- Track selection does not have an effect

I am still trying to find a way to predictably make this happen and will post back if I come across anything interesting.

**Comment by JC3** [comment]
> Your project is much smaller than mine - might that have some bearing ???

I don't think it does; it happens for me over a very wide range of project complexities and sizes.

**Comment by JC3** [comment]
I still can't figure out what triggers this but just giving this a little bump since I also have to keep switching back to 2.x when I need to do any serious editing.
